course/layouts/partials/learning_resource_types.html

```diff
@@ -1,14 +1,17 @@
 {{ $courseData := .context.Site.Data.course }}
 {{ $inPanel := .inPanel }}
-<div class="course-home-section w-100 {{ if $inPanel }}in-panel{{ else }}bg-light{{ end }}">
-  <div class="container px-0 mx-0">
-    <h4 class="course-info-title font-weight-bold">Learning Resource Types</h4>
-    <div class="row">
-      {{ range $courseData.learning_resource_types }}
-        <div class="{{ if $inPanel }}col-12{{ else }}col-6 col-xl-3{{ end }}">
-          {{ partial "learning_resource_type.html" . }}
-        </div>
-      {{ end }}
+
+{{if gt (len $courseData.learning_resource_types) 0}}
+  <div class="course-home-section w-100 {{ if $inPanel }}in-panel{{ else }}bg-light{{ end }}">
+    <div class="container px-0 mx-0">
+      <h4 class="course-info-title font-weight-bold">Learning Resource Types</h4>
+      <div class="row">
+        {{ range $courseData.learning_resource_types }}
+          <div class="{{ if $inPanel }}col-12{{ else }}col-6 col-xl-3{{ end }}">
+            {{ partial "learning_resource_type.html" . }}
+          </div>
+        {{ end }}
+      </div>
     </div>
   </div>
-</div>
+{{end}}
```

Comment:
Looking good just one small change.
we just need to wrap this in another if condition or use the `and `operator to check if the  `$courseData.learning_resource_types` exists.
That way if  someone removes the learning_resource_type from course.json it will still work

Reply:
I'm not sure whether this is required as if $courseData.learning_resource_types does not exist then it will break in other places too. Even if we have to place a check for checking existence then I guess it should be placed in other files too and for other attributes too for better consistency. 
Maybe this can be done somewhere which is more generic or maybe we can do it here.
Perhaps @gumaerc can share his thoughts?